src/main/java/com/aws/greengrass/deployment/DefaultDeploymentTask.java

```diff
@@ -198,6 +199,14 @@ public DeploymentResult call() throws InterruptedException {
             if (isLocalDeployment && ThingGroupHelper.DEVICE_OFFLINE_INDICATIVE_EXCEPTIONS.contains(e.getClass())) {
                 logger.atWarn().setCause(e).log("Failed to get thing group hierarchy, local deployment will proceed");
                 groupsForDeviceOpt = getPersistedMembershipInfo();
+            } else if (e instanceof GreengrassV2DataException && ((GreengrassV2DataException) e).statusCode() == 403) {
```

Comment:
positive that this is the exception type? May want to use the super class (SdkServiceException?)

Reply:
or get more specific and use AccessDeniedException